boxes/generators/cardbox.py

```diff
@@ -0,0 +1,153 @@
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from boxes import edges, Boxes
+
+
+class InsetEdgeSettings(edges.Settings):
+    """Settings for InsetEdge"""
+    absolute_params = {
+        "thickness": 0,
+    }
+
+
+class InsetEdge(edges.BaseEdge):
+    """An edge with space to slide in a lid"""
+    def __call__(self, length, **kw):
+        t = self.settings.thickness
+        self.corner(90)
+        self.edge(t, tabs=2)
+        self.corner(-90)
+        self.edge(length, tabs=2)
+        self.corner(-90)
+        self.edge(t, tabs=2)
+        self.corner(90)
+
+
+class FingerHoleEdgeSettings(edges.Settings):
+    """Settings for FingerHoleEdge"""
+    absolute_params = {
+        "wallheight": 0,
+    }
+
+
+class FingerHoleEdge(edges.BaseEdge):
+    """An edge with room to get your fingers around cards"""
+    def __call__(self, length, **kw):
+        depth = self.settings.wallheight-self.thickness*2-10
+        self.edge(length/2-10, tabs=2)
+        self.corner(90)
+        self.edge(depth, tabs=2)
+        self.corner(-180, 10)
+        self.edge(depth, tabs=2)
+        self.corner(90)
+        self.edge(length/2-10, tabs=2)
+
+
+class CardBox(Boxes):
+    """Box for storage of playingcards"""
+    ui_group = "Box"
+
+    def __init__(self):
+        Boxes.__init__(self)
+        self.addSettingsArgs(edges.FingerJointSettings)
+        self.buildArgParser(h=30)
+        self.argparser.add_argument(
+            "--cardwidth",  action="store", type=float, default=65,
+            help="Width of the cards")
+        self.argparser.add_argument(
+            "--cardheight",  action="store", type=float, default=90,
+            help="Height of the cards")
+        self.argparser.add_argument(
+            "--lipwidth",  action="store", type=float, default=10,
+            help="Width of the top lips")
+
+    @property
+    def boxwidth(self):
+        return self.cardwidth*2 + self.thickness*3
+
+    def divider_bottom(self):
+        x = self.boxwidth
+        y = self.cardheight
+
+        self.fingerHolesAt(x/2, 0, y, 90)
+
+    def divider_back_and_front(self):
+        x = self.boxwidth
+        y = self.h - self.thickness
+
+        self.fingerHolesAt(x/2, 0, y, 90)
+
+    def render(self):
+        self.open()
+        h = self.h
+        t = self.thickness
+
+        x = self.boxwidth
+        y = self.cardheight
+
+        s = InsetEdgeSettings(thickness=t)
+        p = InsetEdge(self, s)
+        p.char = "a"
+        self.addPart(p)
+
+        s = FingerHoleEdgeSettings(thickness=t, wallheight=h)
+        p = FingerHoleEdge(self, s)
+        p.char = "A"
+        self.addPart(p)
+
+        self.ctx.save()
+
+        # Lid
```

Comment:
The lid is a bit too wide. It is a press fit as all parts are by default. The width need to be reduced to allow some play and easy movement.0.1*t or even 0.2*t may be a good value.
At the same time the lid is not deep enough. It should overlap the over wall. So it needs to be y+t deep.
I would probably also add a finger hole to make sliding the lid out easier.

Reply:
Having the back (or all) corners of the lid rounded with a small radius would make inserting the lid much easier.